### ISSUE TITLE
Global state missing after crash recovery

### ISSUE THREAD

**Issue by ichinaski** [akvo/akvo-flow-mobile#258]
Upon an unforeseen app crash, the Android environment attempts to recover the previous state and ongoing activities. However, if the FLOW app is set not to automatically log in the last user, this part of the global state will not be properly loaded.

Since most of the activities assume the user is logged in (except main screen), this can potentially result in an unexpected NPE -- blindly assuming the `User` object can be read from the _FlowApp_ component.

To go around this situation, we'll need to explicitly load the user after an app crash.

_Note: Obviously the crash should not happen in the first place, but we must relent this behaviour while we find potential scenarios whereby the app could crash_.

**Comment by ichinaski** [comment]
I tried to check `null` references on *RecordActivity*'s `onResume()` callback. Although this check did work and the activity can simply explicitly finish the life-cycle at this point, I realised some other actvities may suffer the same problem. As this solution was meant to be a temporary thing, and the cost of executing it (code maintainability-wise) is higher than expected, we'll have to take a better, more longsighted approach.
